src/utils/urlUtils.js

```diff
@@ -72,24 +73,48 @@ export const handleSingleURL = async (url, articleDataList, setMessageList) => {
     return null;
   }
 
-  const articleData = await requestURL(url);
-  const { statusCode } = articleData;
-
-  if (statusCode !== 200) {
-    setMessageList((prev) => [
-      ...prev,
-      {
-        id: crypto.randomUUID(),
-        icon: ERROR_MESSAGE[statusCode].icon,
-        messages: ERROR_MESSAGE[statusCode].messages,
-        url,
-      },
-    ]);
-
-    return null;
+  const controlloer = new AbortController();
+  const controlloerOption = {
+    signal: controlloer.signal,
+  };
+  const fetchTimer = setTimeout(() => {
+    controlloer.abort();
+  }, 3000);
+
+  try {
+    const articleData = await requestURL(url, controlloerOption);
+    const { statusCode } = articleData;
+
+    if (statusCode !== 200) {
+      setMessageList((prev) => [
+        ...prev,
+        {
+          id: crypto.randomUUID(),
+          icon: ERROR_MESSAGE[statusCode].icon,
+          messages: ERROR_MESSAGE[statusCode].messages,
+          link: url,
+        },
+      ]);
+    } else {
+      return articleData.data;
+    }
+  } catch (error) {
+    if (error.name === "AbortError") {
+      setMessageList((prev) => [
+        ...prev,
+        {
+          id: crypto.randomUUID(),
+          icon: ERROR_MESSAGE[408].icon,
+          messages: ERROR_MESSAGE[408].messages,
+          link: url,
+        },
+      ]);
+    }
+  } finally {
```

Comment:
```suggestion
    if (error.name === "AbortError") {
      setMessageList((prev) => [
        ...prev,
        {
          id: crypto.randomUUID(),
          icon: ERROR_MESSAGE[408].icon,
          messages: ERROR_MESSAGE[408].messages,
          link: url,
        },
      ]);
    } else {
      throw error;
    }
  } finally {
```

error.name === "AbortError"가 아닌 경우 error를 throw하는 것이 어떨까요?